visualeyes/core/plotting.py

```diff
@@ -1,99 +1,79 @@
+    
+    screen_dimension = (aoi_mask.shape[1], aoi_mask.shape[0])
 
-    # start an empty list to save ratio of fixations within the aoi for each trial
-    ratios_fixations_within = []
+    visualeyes.aoi_mask_validation(aoi_mask, screen_dimension)
 
-    for row_idx, trial in df_trials.iterrows():
-        print(f'trial indx: {row_idx}')
-        trial_stime = trial['stime']
-        trial_etime = trial['etime']
+    for index in epoch_data['epoch_index'].unique():
+        epoch_x = x_coord[x_coord['epoch_index'] == index]
+        epoch_y = y_coord[y_coord['epoch_index'] == index]
 
-        # locate the fixations within trial start and end time
-        fixations_in_trial = df_fixations[(df_fixations['stime'] >= trial_stime) & (df_fixations['etime'] <= trial_etime)]
+        out_of_bounds = any(x < 0 or x >= aoi_mask.shape[1] or y < 0 or y >= aoi_mask.shape[0]
+                           for x,y in zip(epoch_x, epoch_y)
+                           )
 
-        # find the x and y coordinates of fixations 
-        x_coor = fixations_in_trial ['axp']
-        y_coor = fixations_in_trial ['ayp']
+        if out_of_bounds:
+            print(f"Some coordinates for epoch {index} are out of bounds")
 
-        # count the number of fixations
-        num_fixations = x_coor.shape[0]
+        else:
 
-        # generate the binary image as aoi mask
-        edf_file = read_edf(filepath)
-        
-        # extract screen coordinates from the 'info' column of the edf file
-        screen_width = edf_file['info']['screen_coords'][0]
-        screen_height = edf_file['info']['screen_coords'][1]
-
-        # define boundaries of the aoi (assume rectangle shape)
-        SHAPE = 'rectangle'
-        PIXELS = 100
-        x1 = screen_width/2 - PIXELS
-        x2 = screen_width/2 + PIXELS
-        y1 = screen_height/2 - PIXELS
-        y2 = screen_height/2 + PIXELS
-        
-        # initiate aoi parameters dict
-        aoi_parameters = [{'shape': SHAPE, 'coordinates': (x1, y1, x2, y2)}]
+            plt.imshow(aoi_mask, cmap='gray')
+
+            plt.title(f"plot epoch {index} data on AOI mask")
+
+            plt.set_xlim(0, aoi_mask.shape[1]-1)
 
-        # generate aoi mask
-        aoi_mask = define_aoi(screen_width, screen_height, aoi_parameters)
+            plt.set_ylim(0, aoi_mask.shape[0]-1)
 
-        plt.imshow(aoi_mask, cmap='gray')
-        plt.title("Fixations per trial on the AOI mask")
-        plt.gca().set_ylim(plt.gca().get_ylim()[::-1]) # invert the y-axis to align with eyelink
-        plt.gca().xaxis.tick_top()
-        plt.scatter(x_coor, y_coor, color='skyblue', marker='x', s=15) # plot fixations within trial
-        plt.savefig(os.path.join(savepath, f'Fixations_in_AOI_{row_idx+1}.svg')) # save the svg for each trial in the savepath
+            plt.gca().set_ylim(plt.gca().get_ylim()[::-1]) # invert the y-axis to align with eyelink dataviewer origins and coordinates

```

Comment:
I mentioned this approach when we discussed the function, but on second thought this is not the best approach. Sometimes it's arbitrary what the "current axis" is. Instead, consider doing:
```python
fig, ax = plt.subplots() # to intentionally initialize a fig and ax objects
```
This intentionally creates `fig` and `ax` object to avoid confusion. Additionally, the function can also take a user defined `ax` parameter to give people more control on the plot outside the function.

Reply:
And to invert axis, you can just do `ax.invert_yaxis` and `ax.invert_xaxis`